app/controllers/reports_controller.rb

```diff
@@ -54,6 +54,18 @@ def selection
 
   def review
     @records = Record.where(id: params[:record_id])
+    @granule_associations = {}
+    @records.each do |record|
+      value = record.associated_granule_value
+      if value.nil? || value.empty? || value == 'Undefined'
```

Comment:
You can leave this as is, but if the `.empty?` check is not needed I think it can be removed.